adadamp/_dist.py

```diff
@@ -74,127 +99,154 @@ def gradient(
+        warn(f"Model appears not to update with diff={rel_error}")
 
-    This class has a Scikit-learn API. It accepts PyTorch datasets and
-    array-like objects (PyTorch Tensors or NumPy arrays).
+    return model, optimizer
 
-    Parameters
-    ----------
-    module : torch.nn.Module
-    loss : PyTorch Loss
-    optimizer : torch.optim.Optimizer
-    batch_size : int
-        The batch size to use.
-    max_batch_size : int, optional (default: 4096).
-        The maximum batch size to use.
-    min_workers : int
-        The minimum number of workers available
-    max_workers : int
-        The maximum number of workers available
-    device : str
-        The device used. Directly passed to :class:`torch.device`.
-    grads_per_worker: int
-        How many gradients should each worker handle?
-    max_epochs : int
-        The maximum number of passes through the dataset (or epochs) to train
-        for.
-    **kwargs : Dict[str, Any]
-        Keyword arguments to pass to Skorch. For example, these can include
-        ``optimizer__lr`` to set the learning rate or ``module__foo`` to pass
-        keyword argument ``foo`` to module initialization.
-
-    Attributes
-    ----------
-    module_ : nn.Module
-        The initialize PyTorch module.
-    optimizer_ : Optimizer
-        The initialize PyTorch optimizer.
-    loss_ : Loss
-        The initialized PyTorch loss.
-    initialized_ : bool
-        Whether this class has been initialized.
-    meta_ : Dict[str, Number]
-        Meta information about the fitting process. This includes keys
-        ``['n_updates', 'n_data', 'score__calls', 'partial_fit__calls']``.
-    """
+
+class DaskBaseDamper:
     def __init__(
         self,
-        module: nn.Module,
-        loss: Loss,
-        optimizer: Optimizer,
+        module,
+        loss,
+        optimizer,
         *,
-        batch_size: int=32,
-        max_batch_size: Optional[int] = 4096,
-        min_workers: int=1,
-        max_workers: int=8,
+        metrics=None,
+        batch_size=32,
+        cluster=None,
+        max_batch_size=None,
+        lr=1,
+        worker_max_batch_size=1024,
+        min_workers=1,
+        max_workers=8,
         device: str = "cpu",
-        grads_per_worker: int=128,
-        max_epochs: int=20,
+        grads_per_worker=32,
+        max_epochs=20,
+        random_state=None,
         **kwargs,
     ):
         self.module = module
         self.loss = loss
         self.optimizer = optimizer
+        self.metrics = metrics
         self.device = device
+        self.cluster = cluster
         self.grads_per_worker = grads_per_worker
         self.max_epochs = max_epochs
+        self.lr = lr
 
         self.batch_size = batch_size
+        self.max_batch_size = max_batch_size
+        self.worker_max_batch_size = worker_max_batch_size
         self.min_workers = min_workers
         self.max_workers = max_workers
+        self.n_workers_ = min_workers
```

Comment:
This should be set in `initialize`.

The Scikit-learn API says that parameters ending in underscores should only be set when `fit` is called (https://scikit-learn.org/stable/glossary.html#term-attributes)